src/scripts.js

```diff
@@ -142,40 +140,98 @@ function buildTags(recipe) {
     .join(' ');
 }
 
+function displayAllTags() {
+  let allTags;
+
+  if (allRecipesTab.checked) {
+    allTags = recipeRepo.retrieveAllTags();
+  }
+  else if (!allRecipesTab.checked) {
+    allTags = user.favoriteRecipeRepo.retrieveAllTags();
+  }
+
+  allTags.forEach((tag) => {
+    const tagElement = document.createElement('p');
+    tagElement.classList.add('recipe-tag');
+    tagElement.innerText = tag;
+
+    tagElement.onclick = tagsToggleFilter;
+
+    tagsContainer.appendChild(tagElement);
+    tagList.push(tagElement);
+  });
+}
+
 function tagsToggleFilter(event) {
   if (tag === event.target.innerText) {
-    updateAllRecipeDisplay(recipeRepo.allRecipes);
-    event.target.classList.remove('recipe-tag-selected');
-    tag = '';
+    removeTag(event)
   } else {
-    const filterSelectedTags = tagList.filter((tag) =>
-      tag.classList.contains('recipe-tag-selected')
-    );
-    filterSelectedTags.forEach((tag) =>
-      tag.classList.remove('recipe-tag-selected')
-    );
-
-    let userTag = getTag(event);
+    addTag(event)
+  }
+}
+
+function addTag(event) {
+  const filterSelectedTags = tagList.filter((tag) =>
+    tag.classList.contains('recipe-tag-selected')
+  );
+
+  filterSelectedTags.forEach((tag) =>
+    tag.classList.remove('recipe-tag-selected')
+  );
+
+  let userTag = getTag(event);
+
+  if (allRecipesTab.checked) {
     const filteredRecipes = recipeRepo.filterByTag(userTag);
-    updateAllRecipeDisplay(filteredRecipes);
+    updateRecipeDisplay(filteredRecipes);
+  }
+  else if (cookbookTab.checked) {
+    const filteredRecipes = user.favoriteRecipeRepo.filterByTag(userTag);
+    updateRecipeDisplay(filteredRecipes);
```

Comment:
Great logic and separation of DOM and data model to display all recipes vs cookbook